shared-module/pathlib/PosixPath.c

```diff
@@ -0,0 +1,333 @@
+            last_slash = i;
+            break;
+        }
+    }
+
+    // No slash found, return "."
+    if (last_slash == -1) {
+        return pathlib_posixpath_from_str(MP_OBJ_NEW_QSTR(MP_QSTR__dot_));
+    }
+
+    // Root directory
+    if (last_slash == 0) {
+        return pathlib_posixpath_from_str(MP_OBJ_NEW_QSTR(MP_QSTR__slash_));
+    }
+
+    // Create parent path
+    mp_obj_t parent = mp_obj_new_str((const char *)path_str, last_slash);
+    return pathlib_posixpath_from_str(parent);
+}
+
+mp_obj_t common_hal_pathlib_posixpath_name(pathlib_posixpath_obj_t *self) {
+    GET_STR_DATA_LEN(self->path_str, path_str, path_len);
+
+    // Empty path
+    if (path_len == 0) {
+        return MP_OBJ_NEW_QSTR(MP_QSTR_);
+    }
+
+    // Find the last slash
+    int last_slash = -1;
+    for (int i = path_len - 1; i >= 0; i--) {
+        if (path_str[i] == '/') {
+            last_slash = i;
+            break;
+        }
+    }
+
+    // No slash found, return the whole path
+    if (last_slash == -1) {
+        return self->path_str;
+    }
+
+    // Slash at the end
+    if (last_slash == (int)(path_len - 1)) {
+        return MP_OBJ_NEW_QSTR(MP_QSTR_);
+    }
+
+    // Return the component after the last slash
+    return mp_obj_new_str((const char *)(path_str + last_slash + 1), path_len - last_slash - 1);
+}
+
+mp_obj_t common_hal_pathlib_posixpath_stem(pathlib_posixpath_obj_t *self) {
+    mp_obj_t name = common_hal_pathlib_posixpath_name(self);
+    GET_STR_DATA_LEN(name, name_str, name_len);
+
+    // Empty name
+    if (name_len == 0) {
+        return name;
+    }
+
+    // Find the last dot
+    int last_dot = -1;
+    for (int i = name_len - 1; i >= 0; i--) {
+        if (name_str[i] == '.') {
+            last_dot = i;
+            break;
+        }
+    }
+
+    // No dot found or dot at the start, return the whole name
+    if (last_dot <= 0) {
+        return name;
+    }
+
+    // Return the part before the last dot
+    return mp_obj_new_str((const char *)name_str, last_dot);
+}
+
+mp_obj_t common_hal_pathlib_posixpath_suffix(pathlib_posixpath_obj_t *self) {
+    mp_obj_t name = common_hal_pathlib_posixpath_name(self);
+    GET_STR_DATA_LEN(name, name_str, name_len);
+
+    // Empty name
+    if (name_len == 0) {
+        return MP_OBJ_NEW_QSTR(MP_QSTR_);
+    }
+
+    // Find the last dot
+    int last_dot = -1;
+    for (int i = name_len - 1; i >= 0; i--) {
+        if (name_str[i] == '.') {
+            last_dot = i;
+            break;
+        }
+    }
+
+    // No dot found or dot at the start, return empty string
+    if (last_dot <= 0) {
+        return MP_OBJ_NEW_QSTR(MP_QSTR_);
+    }
+
+    // Return the part after the last dot (including the dot)
+    return mp_obj_new_str((const char *)(name_str + last_dot), name_len - last_dot);
+}
+
+mp_obj_t common_hal_pathlib_posixpath_exists(pathlib_posixpath_obj_t *self) {
+    if (path_exists(mp_obj_str_get_str(self->path_str))) {
+        return mp_const_true;
```

Comment:
Path('.').exists() was failing for the root folder (and mounted root '/sd') but by using the resolved path the '.' is converted to an absolute path which works. I think this partially resolves the issue of parsing up through sd mount points as well.

```py
    if (path_exists(mp_obj_str_get_str(((pathlib_posixpath_obj_t*)common_hal_pathlib_posixpath_resolve(self))->path_str))) {
```